lib/Zonemaster/CLI.pm

```diff
@@ -407,6 +407,39 @@ sub run {
         Zonemaster::Engine::Profile->effective->merge( $profile );
     }
 
+    my @testing_suite;
+    if ( $self->test and @{ $self->test } > 0 ) {
+        foreach my $t ( @{ $self->test } ) {
+            # There should be at most one slash character
+            if ( $t =~ tr/\/// > 1 ) {
+                die __( "Error: --test must have at most one slash ('/') character: $t");
+            }
+
+            # The case does not matter
+            $t = lc( $t );
+
+            my ( $module, $method );
+            # Fully qualified module and test case (e.g. Example/example12), or just a test case (e.g. example12). Note the different capturing order.
+            if ( ( ($module, $method) = $t =~ m#^ ( [a-z]+ ) / ( [a-z]+[0-9]{2} ) $#ix )
+                or
+                 ( ($method, $module) = $t =~ m#^ ( ( [a-z]+ ) [0-9]{2} ) $#ix ) )
+            {
+                push @testing_suite, "$module/$method";
+                if ( not grep( /^$method$/, @{ Zonemaster::Engine::Profile->effective->get( 'test_cases' ) } ) ) {
+                    # Actual forcing will be done later in the code, i.e. just before calling Zonemaster::Engine->test_method()
+                    say $fh_diag __x( "Notice: Engine does not have test case '$method' enabled in the profile. Forcing...");
+                }
+            }
+            # Just a module name (e.g. Example) or something invalid.
+            # TODO: in case of invalid input, print a proper error message
+            # suggesting to use --list-tests for valid choices.
+            else {
+                $t =~ s{/$}{};
+                push @testing_suite, $t;
```

Comment:
Maybe validate the existence of $module in Zonemaster::Engine->all_methods before adding it to @testing_suite?

Also, have you considered expanding the module name to its constituent method names here (using Zonemaster::Engine->all_methods) and adding those to @testing_suite instead? That way the test_cases property could be set to @testing_suite once and for all just after the `foreach my $t` loop (instead of setting it inside the loop iterating over @testing_suite).